skrifa/src/color/mod.rs

```diff
@@ -0,0 +1,235 @@
+    font: &impl TableProvider<'a>,
+    glyph_id: GlyphId,
+) -> Result<bool, ReadError> {
+    let colr = font.colr()?;
+    Ok(colr.v1_base_glyph(glyph_id)?.is_some())
+}
+
+pub fn get_clipbox<'a>(
+    font: &impl TableProvider<'a>,
+    glyph_id: GlyphId,
+    instance: LocationRef,
+    size: Option<f32>,
+) -> Result<Option<BoundingBox<f32>>, ReadError> {
+    let colr = font.colr()?;
+    let upem = font.head()?.units_per_em();
+    let color_instance = ColrInstance::new(colr, instance.coords());
+
+    let resolved_bounding_box = get_clipbox_font_units(&color_instance, glyph_id)?;
+
+    let scaled_clipbox = resolved_bounding_box.map(|bounding_box| {
+        let scale_factor = size.map(|size| size / upem as f32).unwrap_or(1.0);
+        BoundingBox {
+            x_min: bounding_box.x_min * scale_factor,
+            y_min: bounding_box.y_min * scale_factor,
+            x_max: bounding_box.x_max * scale_factor,
+            y_max: bounding_box.y_max * scale_factor,
+        }
+    });
+    Ok(scaled_clipbox)
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::prelude::LocationRef;
+    use read_fonts::FontRef;
+
+    use super::{
+        has_colrv1_glyph, traverse_color_outline, Brush, ColorPainter, CompositeMode, GlyphId,
+        Transform,
+    };
+
+    #[test]
+    fn has_colrv1_glyph_test() {
+        let colr_font = font_test_data::COLRV0V1_VARIABLE;
+        let font = FontRef::new(colr_font).unwrap();
+        assert!(!has_colrv1_glyph(&font, GlyphId::new(166)).unwrap());
+        assert!(has_colrv1_glyph(&font, GlyphId::new(167)).unwrap());
+    }
+    struct DummyColorPainter {}
+
+    impl DummyColorPainter {
+        pub fn new() -> Self {
+            Self {}
+        }
+    }
+
+    impl Default for DummyColorPainter {
+        fn default() -> Self {
+            Self::new()
+        }
+    }
+
+    impl ColorPainter for DummyColorPainter {
+        fn push_transform(&mut self, _transform: Transform) {}
+        fn pop_transform(&mut self) {}
+        fn push_clip_glyph(&mut self, _glyph: GlyphId) {}
+        fn push_clip_rectangle(&mut self, _x_min: f32, _x_max: f32, _y_min: f32, _y_max: f32) {}
+        fn pop_clip(&mut self) {}
+        fn fill(&mut self, _brush: Brush) {}
+        fn push_layer(&mut self, _composite_mode: CompositeMode) {}
+        fn pop_layer(&mut self) {}
+    }
+
+    #[test]
+    fn paintcolrglyph_cycle_test() {
+        let colr_font = font_test_data::COLRV0V1_VARIABLE;
+        let font = FontRef::new(colr_font).unwrap();
+        let paint_colr_glyph_cycle_gid = GlyphId::new(176);
+        assert!(has_colrv1_glyph(&font, paint_colr_glyph_cycle_gid).unwrap());
+        let mut color_painter = DummyColorPainter::new();
+
+        assert!(traverse_color_outline(
+            &font,
+            paint_colr_glyph_cycle_gid,
+            LocationRef::default(),
+            &mut color_painter,
+        )
+        .is_err());
+    }
+}
+
+#[cfg(all(test, feature = "serde"))]
+pub mod test_glyph_defs;
```

Comment:
This feels like it should be owned by the traversal_tests module. I’d suggest a new traversal _tests directory under color to group these. 

Also, there’s a general expectation that all mod declarations are grouped near the top of the file.